File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {

    public String encode(List<String> strs) {
        String result = new String();
        String delimit = "#";

        for (String string : strs){
            int length = string.length();
            String encode = String.valueOf(length) + delimit + string;
            result += encode;
        }

        return result;
    }

    public List<String> decode(String str) {
        List<String> result = new ArrayList<>();
        
        int i = 0;        
        while (i < str.length()) {
            // 1. Find the delimiter '#'
            int j = str.indexOf('#', i); 
        
            // 2. Parse the FULL length (handles "10#...", not just "1#...")
            int len = Integer.parseInt(str.substring(i, j));
        
            // 3. Extract the string starting AFTER the '#'
            int start = j + 1;
            int end = start + len;
            result.add(str.substring(start, end));
        
            // 4. Update 'i' to the start of the next segment
            i = end;
        }   
        
        return result;
    }
}

File: Data Structures & Algorithms/confusing-number/submission-1.cpp
class Solution {
public:
    bool confusingNumber(int n) {
        map<int, int> invert_map = {{0, 0}, {1, 1}, {6, 9}, {8, 8}, {9, 6}};
        int rotated_num = 0;
        int n_copy = n;

        while (n_copy > 0) {
            int result = n_copy % 10;
            if (invert_map.find(result) == invert_map.end()) {
                return false;
            }

            rotated_num = rotated_num * 10 + invert_map[result];
            n_copy /= 10;
        }

        return rotated_num != n;
    }
};
